src/main/java/org/cyclonedx/util/ExtensionDeserializer.java

```diff
@@ -0,0 +1,255 @@
+          break;
+        case "ratings":
+          vuln.setRatings(processRatings(vulnJson.get(field)));
+          break;
+        case "cwes":
+          vuln.setCwes(processCwes(vulnJson.get(field)));
+          break;
+        case "description":
+          vuln.setDescription(vulnJson.get(field).textValue());
+          break;
+        case "recommendations":
+          vuln.setRecommendations(processRecommendations(vulnJson.get(field)));
+          break;
+        case "advisories":
+          vuln.setAdvisories(processAdvisories(vulnJson.get(field)));
+          break;
+        default:
+          // Can't do anything so don't do anything
+          break;
+      }
+    }
+    return vuln;
+  }
+
+  private List<Advisory> processAdvisories(final JsonNode advisories) {
+    List<Advisory> advisoryList = new ArrayList<>();
+    JsonNode adv = advisories.get("advisory");
+    if (adv.isArray() && !adv.isEmpty()) {
+      for (JsonNode a : adv) {
+        advisoryList.add(processAdvisory(a));
+      }
+    } else {
+      advisoryList.add(processAdvisory(adv));
+    }
+    return advisoryList.isEmpty() ? null : advisoryList;
+  }
+
+  private Advisory processAdvisory(final JsonNode advisory) {
+    Advisory adv = new Advisory();
+    adv.setText(advisory.textValue());
+    return adv;
+  }
+
+  private List<Recommendation> processRecommendations(final JsonNode recommendations) {
+    List<Recommendation> recommendationsList = new ArrayList<>();
+    JsonNode rec = recommendations.get("recommendation");
+    if (rec.isArray() && !rec.isEmpty()) {
+      for (JsonNode r : rec) {
+        recommendationsList.add(processRecommendation(r));
+      }
+    } else {
+      recommendationsList.add(processRecommendation(rec));
+    }
+    return recommendationsList.isEmpty() ? null : recommendationsList;
+  }
+
+  private Recommendation processRecommendation(final JsonNode recommendation) {
+    Recommendation rec = new Recommendation();
+    rec.setText(recommendation.textValue());
+    return rec;
+  }
+
+  private List<Cwe> processCwes(final JsonNode cwes) {
+    List<Cwe> cweList = new ArrayList<>();
+    JsonNode cwe = cwes.get("cwe");
+    if (cwe.isArray() && !cwe.isEmpty()) {
+      for (JsonNode c : cwe) {
+        cweList.add(processCwe(c));
+      }
+    } else {
+      processCwe(cwe);
+    }
+    return cweList.isEmpty() ? null : cweList;
+  }
+
+  private Cwe processCwe(final JsonNode cwe) {
+    Cwe c = new Cwe();
+    c.setText(Integer.valueOf(cwe.textValue()));
+    return c;
+  }
+
+  private Vulnerability1_0.Source processSource(final JsonNode source) {
+    Source sauce = new Source();
+    sauce.setName(getAsString("name", source));
+    if (source.get("url") != null) {
+      try {
+        sauce.setUrl(new URL(source.get("url").textValue()));
+      }
+      catch (MalformedURLException e) {
+        // Should we throw an exception? Is this worth stopping things over?
```

Comment:
IMO I think we should throw the exception, but I think the URL is optional, so need to check what's the best option here

Reply:
@stevespringett I'm at a loss of what to do here. I sorta feel like we should swallow the exception, and move on, but I could also see a reason to like go NOPE CANT GENERATE THE WHOLE SBOM, which is I think what would happen if I throw the exception all the way up the chain. What do you think?